qiskit_experiments/curve_analysis/visualization/base_drawer.py

```diff
@@ -0,0 +1,257 @@
+        Args:
+            fields: The fields to update the options
+        """
+        self._options.update_options(**fields)
+        self._set_options = self._set_options.union(fields)
+
+    @abstractmethod
+    def initialize_canvas(self):
+        """Initialize the drawing canvas."""
+
+    @abstractmethod
+    def format_canvas(self):
+        """Final cleanup for the canvas appearance."""
+
+    @abstractmethod
+    def draw_raw_data(
+        self,
+        x_data: Sequence[float],
+        y_data: Sequence[float],
+        ax_index: Optional[int] = None,
+        **options,
+    ):
+        """Draw raw data.
+
+        Args:
+            x_data: X values.
+            y_data: Y values.
+            ax_index: Index of canvas if multiple inset axis exist.
+            options: Valid options for the drawer backend API.
+        """
+
+    @abstractmethod
+    def draw_formatted_data(
+        self,
+        x_data: Sequence[float],
+        y_data: Sequence[float],
+        y_err_data: Sequence[float],
+        name: Optional[str] = None,
+        ax_index: Optional[int] = None,
+        **options,
+    ):
+        """Draw formatted data that used for fitting.
+
+        Args:
+            x_data: X values.
+            y_data: Y values.
+            y_err_data: Standard deviation of Y values.
+            name: Name of this curve.
+            ax_index: Index of canvas if multiple inset axis exist.
+            options: Valid options for the drawer backend API.
+        """
+
+    @abstractmethod
+    def draw_fit_lines(
+        self,
+        fit_function: Callable,
+        signature: List[str],
+        fit_result: FitData,
+        fixed_params: Dict[str, float],
+        ax_index: Optional[int] = None,
+        **options,
+    ):
+        """Draw fit lines.
+
+        Args:
+            fit_function: The function defines a single curve.
+            signature: The fit parameters associated with the function.
+            fit_result: The result of fit.
+            fixed_params: The parameter fixed during the fitting.
+            ax_index: Index of canvas if multiple inset axis exist.
+            options: Valid options for the drawer backend API.
+        """
+
+    @abstractmethod
+    def draw_fit_report(
+        self,
+        analysis_results: List[AnalysisResultData],
+        chisq: Union[float, Dict[str, float]],
+    ):
+        """Draw text box that shows fit reports.
+
+        Args:
+            analysis_results: List of analysis result entries containing fit parameters.
+            chisq: Chi-squared value from the fitting. If this is provided as a dictionary,
+                the key is also shown with the chi-squared value.
+        """
+
+    @property
+    @abstractmethod
+    def figure(self):
+        """Return figure object handler of the canvas object.
+
+        Note that figure and axis might be different plot when a user provide
+        an axis object which is a part of other multiple axis figure.
```

Comment:
This is a bit hard to follow. Can you reformulate?

Reply:
Done in https://github.com/Qiskit/qiskit-experiments/pull/738/commits/bcd8a6ef6680d6725ffdfc0247389ed1b812a4da. This is too specific comment for Matplotlib so I moved it to the mpl drawer rather than in the base drawer.